py3status/modules/mpris.py

```diff
@@ -0,0 +1,382 @@
+
+            for player_name in player_priority:
+                if player_name in players:
+                    players.remove(player_name)
+                    players_prioritized.append(player_name)
+            if '*' in player_priority:
+                players_prioritized += players
+
+        for player_name in players_prioritized:
+            player_state = self._get_state(player_name)
+
+            if player_state == 'Playing':
+                return player_name
+            elif player_state == 'Paused':
+                players_paused.append(player_name)
+            else:
+                players_stopped.append(player_name)
+
+        if players_paused:
+            return players_paused[0]
+        if players_stopped:
+            return players_stopped[0]
+
+        return 'None'
+
+    def _get_text(self, i3s_config):
+        """
+        Get the current metadatas
+        """
+        is_stream = False
+        player = UNKNOWN
+        state = UNKNOWN
+        artist = UNKNOWN
+        album = UNKNOWN
+        length = UNKNOWN_TIME
+        time = UNKNOWN_TIME
+        title = UNKNOWN
+
+        try:
+            player = self._player.Identity
+            (color, state) = self._get_state_format(i3s_config)
+
+            _time_ms = self._player.Position
+            time = _get_time_str(_time_ms)
+
+            metadata = self._player.Metadata
+            if len(metadata) > 0:
+                is_stream = 'file://' not in metadata.get('xesam:url')
+                title = metadata.get('xesam:title') or UNKNOWN
+                album = metadata.get('xesam:album') or UNKNOWN
+
+                if metadata.get('xesam:artist') is not None:
+                    artist = metadata.get('xesam:artist')[0]
+                else:
+                    # we assume here that we playing a video and these types of
+                    # media we handle just like streams
+                    is_stream = True
+
+                _length_ms = metadata.get('mpris:length')
+                if _length_ms is not None:
+                    length = _get_time_str(_length_ms)
+            else:
+                # use stream format if no metadata is available
+                is_stream = True
+        except Exception:
+            color = i3s_config['color_bad']
+
+        if is_stream:
+            # delete the file extension
+            title = re.sub(r'\....$', '', title)
+            format = self.format_stream
+        else:
+            format = self.format
+
+        return (format.format(player=player,
+                              state=state,
+                              album=album,
+                              artist=artist,
+                              length=length,
+                              time=time,
+                              title=title), color)
+
+    def _get_control_states(self):
+        control_states = {
+            'pause':    {'clickable': self._player.CanPause,
```

Comment:
like the action I think it would be nice to make this a string too.  We would only have to check for controls we care about rather than all of them and also this would then be static and would just need to be set up once.

Stop is a funny one as there is no `CanStop` but defining this as `None` would be easy to get around this issue.